subsys/net/ip/net_context.c

```diff
@@ -565,6 +565,9 @@ int net_context_get(sa_family_t family, enum net_sock_type type, uint16_t proto,
 #if defined(CONFIG_NET_IPV4_MAPPING_TO_IPV6)
 		/* By default IPv4 and IPv6 are in different port spaces */
 		contexts[i].options.ipv6_v6only = true;
+#endif
+#if defined(CONFIG_NET_IPV4)
+		contexts[i].ipv4_mcast_loop = true;
```

Comment:
Please take a look of IPv6 PR #87338 and have a Kconfig option to set the initial value.